js/Source/Request/Request.PollForUpdate.js

```diff
@@ -54,6 +55,9 @@ Request.PollForUpdate = new Class({
       this.request = new Request.JSON({
         url: this.url,
         onComplete: this._handleData.bind(this),
+        onFailure: function(){
+          if (this.options.haltOnFailure) this.stop();
+        }
```

Comment:
missing comma.

Would technically be better to only add the event listener if the option is true (instead of evaluating it every time the request replies). I.e. `this.request = new Request(...); if (this.options.haltOnFailure) this.request.addEvent(...);`